tools/generate_syscall_tables.sh

```diff
@@ -0,0 +1,162 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+# This script generates the syscall tables for seccompiler.
+
+set -e
+
+# Full path to the seccompiler tools dir.
+TOOLS_DIR=$(cd "$(dirname "$0")" && pwd)
+
+# Full path to the seccompiler sources dir.
+ROOT_DIR=$(cd "${TOOLS_DIR}/.." && pwd)
+
+# Path to the temporary linux kernel directory.
+KERNEL_DIR="${ROOT_DIR}/.kernel"
+
+generate_syscall_table_x86_64() {
```

Comment:
I think this needs to be reworked to make it easier to understand, and to avoid some of the code duplication.

To do that, we can work backwards from the following responsibilities that our bash script has:
- generate a header file based on the kernel version
- generate a rust code that returns a HashMap based on the platform & the downloaded kernel
- generate a list of syscall name + syscall number based on the kernel file
- write the generated code to a rust file per platform
- format the resulted rust code

From the above list, the only thing that is platform specific is generating the list of syscalls because we need different parsing of different header files. Everything else can be parametrized.

With this in mind, we can come up with the following code structure:
- `generate_syscall_list_x86_64` -> generates the list of x86_64 syscalls as a tuple of &str, i64 (more on why str and not String as a comment in the following commit)
- `generate_syscall_list_aarch64` -> generates the list of aarch64 syscalls as a tuple of &str, i64
- `get_rust_file_header` -> returns the rust header file based on the kernel version
- `get_rust_code` -> generates the rust code that we want to write; this means that it needs to write a Rust function that returns a HashMap (more comments on why returning a HashMap and not append to a mutual parameter in a comment in the following commit); for this function to be able to write the rust code, it will need the syscall table, and the syscall table is platform specific, so this function will take as a parameter the platform for which to generate the code, and then it will call into the appropriate `generate_syscall_list_*` function.
- `write_rust_sycall_table` -> finally, we need a function that does the high level job of writing to a Rust file the header & code by taking into consideration the platform, the kernel version, and the file where the code needs to be written.

Here is how it would look:
```bash
get_rust_code() {
    platform="$1"

    if [ "$platform" == "x86_64" ]; then
      syscall_list=$(generate_syscall_list_x86_64)
    elif [ "$platform" == "aarch64" ]; then
        syscall_list=$(generate_syscall_list_aarch64)
    else
      die "Invalid platform"
    fi

    printf "
    use std::collections::HashMap;
    pub(crate) fn make_syscall_table() -> HashMap<&'static str, i64> {
      [%s].iter().cloned().collect()
    }" "${syscall_list}"
}

write_rust_syscall_table() {
    kernel_version="$1"
    platform="$2"
    rust_file="$3"

    header=$(get_rust_file_header "$kernel_version")
    code=$(get_rust_code "$platform")

    echo "${header}" "${code}" > "${rust_file}"
}
```

The function that needs to be called in main is `write_rust_syscall_table`, and everything will be handled there:

```bash
    validate_kernel_version "$kernel_version"
    # download_kernel "$kernel_version"

    # generate syscall table for x86_64
    echo "Generating table for x86_64..."
    write_rust_syscall_table "$kernel_version" "x86_64" "$path_to_x86_64_table"

    # generate syscall table for aarch64
    echo "Generating table for aarch64..."
    write_rust_syscall_table "$kernel_version" "aarch64" "$path_to_aarch64_table"
```